sim/sample.py

```diff
@@ -4,172 +4,181 @@
 import os
 
 class Sample:
-    per_base_error_rate = "0"
-    num_read_pairs = 150000
-
-    def simulate_genome(self):
-        raise NotImplementedError()
 
+    # Do we need this?
     @property
     def name(self):
         # TODO: this is ugly, make this human-readable
         return f"unnamed-{str(id(self))}"
 
-    def simulate_reads(self, simulated_genome_path, simulated_reads_path):
-        # # TODO: explicitly path fasta path to simulate
-        fasta_path = simulated_genome_path + self.name + '.simulated.simseq.genome.fa'
-        simulate_reads(
-            fasta_path, 
-            simulated_reads_path, 
-            sample_name=self.name, 
-            per_base_error_rate=self.per_base_error_rate,
-            num_read_pairs=self.num_read_pairs
-        )
+    def simulate_genome(self):
+        raise NotImplementedError("Please impliment this method")
+
+    def _simulate_genome_base(self, reference_path, simulated_genome_path, params):
```

Comment:
why does this start with _ ?

Reply:
I opted to make this a protected method because I felt it it doesn't really make sense to call outside of a subclass. Happy to change however.

Reply:
I see, that makes sense, we can leave it with _ if you prefer.